pkg/kclient/generator/generators.go

```diff
@@ -0,0 +1,347 @@
+package generator
+
+import (
+
+	// api resource types
+
+	"fmt"
+
+	routev1 "github.com/openshift/api/route/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	extensionsv1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	devfileParser "github.com/openshift/odo/pkg/devfile/parser"
+	versionsCommon "github.com/openshift/odo/pkg/devfile/parser/data/common"
+)
+
+const (
+	// DevfileSourceVolume is the constant containing the name of the emptyDir volume containing the project source
+	DevfileSourceVolume = "devfile-projects"
```

Comment:
do the other projects follows the concept of source volume?

Reply:
I have asked this question on the forum-devfile Slack channel, lets wait for the outcome. I think all devfile consumers should sync the project somewhere in the pod.

Reply:
I've moved the `odo-projects` vol mount code out of the generators and made it odo specific